metricshub-engine/src/test/java/org/sentrysoftware/metricshub/engine/connector/model/common/EmbeddedFileTest.java

```diff
@@ -0,0 +1,103 @@
+package org.sentrysoftware.metricshub.engine.connector.model.common;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+import org.sentrysoftware.metricshub.engine.common.helpers.MetricsHubConstants;
+
+class EmbeddedFileTest {
+
+	@Test
+	void testDescription() {
+		assertEquals(
+			"EmbeddedFile 1: script.bat",
+			EmbeddedFile.builder().content("value".getBytes()).filename("script.bat").id(1).build().description()

```

Comment:
The name alone is insufficient since different files with the same name can exist in various subdirectories and be referenced in the connector. Therefore, a unique ID is necessary. Additionally, the global index of the embedded files should be represented using integer values instead of strings.